dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -119,6 +119,30 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
     }
   }
 
+  def createNewRangedEvent(
+      eventName: String,
+      maturationTime: Instant,
+      start: Int,
+      stop: Int,
+      step: Int = 1): Future[EventDb] = {
+    if (step == 0) {
+      Future.failed(new IllegalArgumentException("Cannot have a step of 0"))
+    } else {
+      if (step > 0) {
```

Comment:
If you add the comment above, I don't think this is needed anymore.

Also, why must `start < step `? Can't you have 

start=10
step=1
stop=20

Reply:
see comments above and below